### PR TITLE
tests/nested/core20/save: a test that verifies ubuntu-save is present and set up

### PR DESCRIPTION
**Pull request by bboozzoo**
Add a test for checking that ubuntu-save is present and set up correctly.

### CODE REVIEW

File: tests/nested/core20/save/task.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    # transition to recovery mode and check again
+    boot_id="$(nested_get_boot_id)"
+    # shellcheck disable=SC2016
+    nested_exec 'sudo snap reboot --recover $(sudo snap recovery | grep -v Label | awk "{print \$1}")'
+    nested_wait_for_reboot "${boot_id}"
+    # verify in recover mode
+    nested_exec 'sudo cat /proc/cmdline' | MATCH snapd_recovery_mode=recover
+
+    recover_save_out="$(nested_exec "df -B1 /run/mnt/ubuntu-save | tail -1")"
+    echo "$recover_save_out" | MATCH '^/dev/mapper/ubuntu-save-[0-9a-z-]+\s+'
```
Comment (sergiocazzolato) [inline tests/nested/core20/save/task.yaml:35]: should we compare with the previous save_out var? or it is not needed?

**Review by sergiocazzolato — APPROVED** [review]
Thanks for the test, I just left a question inline

File: tests/nested/core20/save/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/nested/core20/save/task.yaml:35]: No, the uuid we add to ubuntu-save name will be different, so we can only compare the size (which should be the same)

**Review by mvo5 — APPROVED** [review]

File: tests/nested/core20/save/task.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    . "$TESTSLIB/nested.sh"
+
+    # check that ubuntu-save is mounted and has a reasonable amount of free
+    # space, example df output:
+    # Filesystem                                                   1B-blocks  Used Available Use% Mounted on
+    # /dev/mapper/ubuntu-save-0bed13ef-f71f-418f-b046-b1ce32dd04a7   5079040 28672   4390912   1% /run/mnt/ubuntu-save
+    save_out="$(nested_exec "df -B1 /run/mnt/ubuntu-save | tail -1")"
+    echo "$save_out" | MATCH '^/dev/mapper/ubuntu-save-[0-9a-z-]+\s+'
+    save_size="$(echo "$save_out" | awk '{print $4}')"
+    echo "check  there is at least 4MB of free space available on ubuntu-save"
```
Comment (anonymouse64) [inline tests/nested/core20/save/task.yaml:18]: nitpick^6000 

```suggestion
    echo "check there is at least 4MB of free space available on ubuntu-save"
```

**Review by anonymouse64 — APPROVED** [review]
LGTM, thanks for the test

**Pull request merged**